modal/cls.py

```diff
@@ -168,20 +195,41 @@ def _initialize_from_other(self, other: "_Cls"):
     def _set_output_mgr(self, output_mgr: OutputManager):
         self._output_mgr = output_mgr
 
+    def _get_partial_functions(self) -> Dict[str, _PartialFunction]:
+        if not self._user_cls:
+            raise AttributeError("You can only get the partial functions of a local Cls instance")
+        return _find_partial_methods_for_user_cls(self._user_cls, _PartialFunctionFlags.all())
+
     def _hydrate_metadata(self, metadata: Message):
+        assert isinstance(metadata, api_pb2.ClassHandleMetadata)
+
+        if metadata.class_function_id:
+            # we don't have any hydration metadata on "service function" themselves
```

Comment:
Yeah it doesn't make sense anymore. I think I used to send in `None` here when I wrote the comment.

Now it simply sends in an empty FunctionHandleMetadata object that it receives from the server instead (the comment still holds in that regard).

Class service functions are `_Function` objects for code organizational reasons, but they don't have/require the regular function handle metadata that other functions have, since they can't have useful values for `web_url`, `function_type` etc. (these attributes belong with the callable i.e. the methods of the class).